language/move-prover/bytecode/src/read_write_set_analysis.rs

```diff
@@ -879,3 +842,67 @@ impl Default for ReadWriteSetState {
         }
     }
 }
+
+// =================================================================================================
+// Converting Infer Result into standalone types
+
+impl Offset {
+    pub fn make_canonical(&self, env: &GlobalEnv) -> Option<RWOffset> {
+        Some(match self {
+            Offset::Field(idx) => RWOffset::Field(*idx),
+            Offset::VectorIndex => RWOffset::VectorIndex,
+            Offset::Global(s) => RWOffset::Global(s.get_type().into_struct_type(env)?),
```

Comment:
TIL you can use `?` with `Option` as well as `Result`!

Reply:
I think I'm using `?` here. Am I missing sth?

Reply:
I'm saying that I didn't know you could do this until I saw you do it!